### PR TITLE
Move unstable is_{arch}_feature_detected! macros to std::arch

### PR DESCRIPTION
**Pull request by Amanieu**
These macros are unstable, except for `is_x86_feature_detected` which is still exported from the crate root for backwards-compatibility.

This should unblock the stabilization of `is_aarch64_feature_detected`.

r? @m-ou-se

### CODE REVIEW

File: library/std/src/lib.rs [hunk 1]
```diff
@@ -527,6 +520,31 @@ pub mod task {
     pub use alloc::task::*;
 }
 
+#[doc = include_str!("../../stdarch/crates/core_arch/src/core_arch_docs.md")]
```
Comment (m-ou-se) [inline library/std/src/lib.rs:523]: The links in the `Other architectures` section are broken when this documentation is put here in `std`. E.g. `x86` links to `std/arch/x86/index.html`, which doesn't exist.

Comment (Amanieu) [reply library/std/src/lib.rs:523]: I think changing the links to point to `../../core/arch/x86/index.html` should fix this?

Comment (m-ou-se) [reply library/std/src/lib.rs:523]: I think so, yes. I was about to say that might break the docs of the `core_arch` crate itself, but it looks like those are already broken in the same way.

**Comment by Amanieu** [comment]
Links are fixed.

**Pull request merged**